### PR TITLE
Ports: Add thesilversearcher (ag)

### PR DESCRIPTION
**Pull request by rwl4**
A faster grep. It's multithreaded and is an order of magnitude faster than grep.

It attempts to utilize LibC's fnmatch which is unimplemented at the moment, but still works using basic string matching.

### CODE REVIEW

File: Ports/thesilversearcher/package.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=thesilversearcher
+version=2.2.0
+useconfigure="true"
+#use_fresh_config_sub=true
```
Comment (linusg) [inline Ports/thesilversearcher/package.sh:5]: Please remove this

**Review by linusg — CHANGES_REQUESTED** [review]
Please squash these commits

File: Ports/AvailablePorts.md [hunk 2]
```diff
@@ -2,226 +2,227 @@
...
+| [`Super-Mario`](Super-Mario/)             | Super-Mario Clone                                               |                          | https://github.com/Bennyhwanggggg/Super-Mario-Clone-Cpp                        |
+| [`tuxracer`](tuxracer/)                   | Tux Racer                                                       | 0.61                     | http://tuxracer.sourceforge.net/                                               |
+| [`tcl`](tcl/)                             | Tcl                                                             | 8.6.12                   | https://www.tcl-lang.org/                                                      |
+| [`termcap`](termcap/)                     | GNU termcap                                                     | 1.3.1                    | https://www.gnu.org/software/termutils/                                        |
+| [`thesilversearcher`](thesilversearcher/) | The Silver Searcher: A fast code-searching tool                 | 2.2.0                    | https://github.com/ggreer/the_silver_searcher                                  |
```
Comment (linusg) [inline Ports/AvailablePorts.md:213]: Ah,, the big reformatting is unfortunate, but no way around it. Could you add a few more spaces (10?) here so the next port with one more character won't have to do the same?

Comment (rwl4) [reply Ports/AvailablePorts.md:213]: Yes, doing that now.

**Pull request merged**